File: 001.c
#include <stdio.h>

int main(void)
{
    int i;
    int sum = 0;

    for (i = 3;  i < 1000; i += 3)  sum += i;
    for (i = 5;  i < 1000; i += 5)  sum += i;
    for (i = 15; i < 1000; i += 15) sum -= i;

    printf("%d\n", sum);
    return 0;
}

File: zig/src/001.zig
const debug = @import("std").debug;

fn run() u32 {
    var i = u32(0);
    var sum = u32(0);
    while (i < 1000) : (i += 1) {
        if (i % 3 == 0) {
            sum += i;
        }
        if (i % 5 == 0) {
            sum += i;
        }
        if (i % 15 == 0) {
            sum -= i;
        }
    }

    return sum;
}

pub fn main() !void {
    const sum = comptime run();
    try debug.warn("{}\n", sum);
}

test "001" {
    debug.assert(run() == 233168);
}
